libs/partners/openai/tests/unit_tests/chat_models/test_base.py

```diff
@@ -334,3 +334,35 @@ def test_with_structured_output(schema: Union[Type[BaseModel], dict]) -> None:
     """Test passing in manually construct tool call message."""
     llm = ChatOpenAI(model="gpt-3.5-turbo-0125", temperature=0)
     llm.with_structured_output(schema)
+
+
+def test_get_num_tokens_from_messages() -> None:
+    llm = ChatOpenAI(model="gpt-4o")
+    messages = [
+        SystemMessage("you're a good assistant"),
+        HumanMessage("how are you"),
+        HumanMessage(
+            [
+                {"type": "text", "text": "what's in this image"},
+                {"type": "image_url", "image_url": {"url": "https://foobar.com"}},
+                {
+                    "type": "image_url",
+                    "image_url": {"url": "https://foobar.com", "detail": "low"},
+                },
+            ]
+        ),
+        AIMessage("a nice bird"),
+        AIMessage(
+            "", tool_calls=[ToolCall(id="foo", name="bar", args={"arg1": "arg1"})]
+        ),
+        AIMessage(
+            "",
```

Comment:
can we try one with non empty text as well? that's the case highlighted in https://github.com/langchain-ai/langchain/issues/23388